File: read_excel.js
const XLSX = require('xlsx');
const path = require('path');

const filePath = path.join(__dirname, 'DAILY_COLLECTION_REPORT_2026-07-28T08-13-25-071Z.xlsx');
const workbook = XLSX.readFile(filePath);

console.log('Sheet Names:', workbook.SheetNames);

for (const sheetName of workbook.SheetNames) {
  const sheet = workbook.Sheets[sheetName];
  const data = XLSX.utils.sheet_to_json(sheet, { header: 1 });
  console.log(`\n=== Sheet: ${sheetName} ===`);
  console.log(`Total rows: ${data.length}`);
  // Print first 20 rows
  for (let i = 0; i < Math.min(20, data.length); i++) {
    console.log(`Row ${i}: ${JSON.stringify(data[i])}`);
  }
}

File: read_excel.py
import subprocess
import sys
subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'openpyxl', '-q'])

from openpyxl import load_workbook

wb = load_workbook('c:/Users/DELL/Documents/report of mpc/DAILY_COLLECTION_REPORT_2026-07-28T08-13-25-071Z.xlsx', data_only=True)
for sheet_name in wb.sheetnames:
    ws = wb[sheet_name]
    print(f'\n=== Sheet: {sheet_name} ===')
    print(f'Rows: {ws.max_row}, Columns: {ws.max_column}')
    for i, row in enumerate(ws.iter_rows(min_row=1, max_row=min(20, ws.max_row), values_only=True)):
        print(f'Row {i+1}: {list(row)}')
